src/mkappimage/mkappimage.go

```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"github.com/probonopd/go-appimage/internal/helpers"
+	"github.com/urfave/cli/v2"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// listFilesInAppImage lists the files in the AppImage, similar to
+// the ls command in UNIX systems
+func listFilesInAppImage(path string) {
+	appimage := NewAppImage(path)
+	err := appimage.ShowContents(false)
+	if err != nil {
+		log.Fatal("Failed to list files: error code:", err)
+	}
+
+}
+
+// listLongFilesInAppImage lists all the files in the appimage
+// similar to `ls -al`. It shows the file permissions also
+func listLongFilesInAppImage(path string) {
+	appimage := NewAppImage(path)
+	err := appimage.ShowContents(true)
+	if err != nil {
+		log.Fatal("Failed to list files: error code:", err)
+	}
+
+}
+
+// bootstrapMkAppImage is a function which converts cli.Context to
+// string based arguments, checks if all the files
+// provided as arguments exists. If yes add the current path to PATH,
+// check if all the necessary dependencies exist,
+// finally check if the provided argument, AppDir is a directly.
+// Call GenerateAppImage with the converted arguments
+// 		Args: c: cli.Context
+func bootstrapMkAppImage(c *cli.Context) error {
+
+	// check if the number of arguments are stictly 1, if not
+	// return
+	if c.NArg() < 1 || c.NArg() > 3 {
+		log.Fatal("Please specify the path to the AppDir/AppImage which you would like to aid.")
+	}
+	fileToAppDir := c.Args().Get(0)
+	fileToAppImageOutput := ""
+	if c.NArg() == 2 {
+		fileToAppImageOutput = c.Args().Get(1)
+	}
+
+	// make sure the output directory exists before continuing
+	outputDirectory := fileToAppImageOutput
+	if strings.HasSuffix(fileToAppImageOutput, ".AppImage") {
+		outputDirectory = filepath.Base(fileToAppImageOutput)
```

Comment:
I'm getting `2021/05/29 13:22:14 The specified output directory does not exist`, though it does. When the target file name does not end with `.AppImage`, this line won't be executed. Thus, the check below will try to assert the the target *file* path is a valid directory, which it of course is not.

Reply:
Why do you want to create an appimage which does not end with `.AppImage`?

Reply:
File extensions aren't really worth a lot, and people might just *do* it. Btw I tested again, and it also doesn't work for me when using the correct file extension (same error message), but I can't see an obvious reason why.

This assumption may also fail in the other direction: suppose somebody accidentally creates a directory with the name of the target AppImage.

Reply:
> File extensions aren't really worth a lot, 

The AppImage principle does not go well with that idea, but I agree with you.

> File extensions aren't really worth a lot, and people might just do it. Btw I tested again, and it also doesn't work for me when using the correct file extension (same error message), but I can't see an obvious reason why.
>
> This assumption may also fail in the other direction: suppose somebody accidentally creates a directory with the name of the target AppImage.

I added a new commit which further refines the checks, and tries to handle the test cases as per your suggestion. 

I'm curious. Do you have a job of writing test cases? :sweat_smile: You did a really good job, and it really makes a lot of sense.

Reply:
> I'm curious. Do you have a job of writing test cases? sweat_smile You did a really good job, and it really makes a lot of sense.

Thank you :) Actually no, just university and free time stuff so far, but I'm starting to work on formal verification (of hardware) soon™ if you let that count ^^
The thing is, I've had to deal with file system foo more than once in the past and sooner or later everything that may go wrong [will blow up in your face](https://en.wikipedia.org/wiki/Murphy%27s_law). At the moment I'm only bothering you with the most basic stuff (i.e. the things that make my cursed use case work), but if you want to we can also have a little talk about symlinks and missing permissions 😈

Reply:
Haha, you are dope!